docs/components/tuner/evaluation.md

```diff
@@ -0,0 +1,190 @@
+# Evaluation
+
+Information retrieval metrics like NDCG and Reciprocal Rank can be calculated using the included 
+{class}`~finetuner.tuner.evaluation.Evaluator` component. The evaluator can be used
+[standalone](#using-the-evaluator)
+or integrated in the training loop via the [evaluation callback](#using-the-evaluation-callback).
+
+
+## Using the evaluator
+
+The evaluator can be used standalone in order to compute the evaluation metrics on a sequence
+of documents:
+```python
+from finetuner.tuner.evaluation import Evaluator
+from docarray import DocumentArray
+
+query_data = DocumentArray(...)
+index_data = DocumentArray(...)
+embed_model = ...
+
+evaluator = Evaluator(query_data, index_data, embed_model)
+
+metrics = evaluator.evaluate()
+```
+
+The `query_data` (or eval data) are the documents that will be evaluated. They can be in both the
+{term}`class dataset` or the {term}`session dataset` format. They should contain ground truths, in the form of
+matches (`doc.matches`) when using session format and in the form of labels when using class format.
+
+If an embedding model is given, the query docs are embedded, otherwise they are assumed to carry
+representations.
+
+The `index_data` (or catalog) is an optional argument that defines the dataset against which the
+query docs are matched. If not provided, query docs are matched against themselves.
+
+The `evaluate()` method returns the computed metrics as a dictionary, mapping metric names to values.
+The computed metrics are the following:
+
+- Precision
+- Recall
+- F1 Score
+- Hit Score
+- R precision
+- Average Precision
+- Reciprocal Rank
+- DCG
+- NDCG
+
+More information on Information Retrieval metrics can be found
+[here](https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)).
+
+Let's consider an example. First, let's build a model that receives a NumPy array with a single element as input, and
+outputs a vector of size 10, with all the elements set to the input scalar:
+```python
+import torch
+
+class EmbeddingModel(torch.nn.Module):
+    
+    @staticmethod
+    def forward(inputs):
+        """
+        input_shape: (bs, 1)
+        output_shape: (bs, 10)
+        """
+        batch_size = inputs.size()[0]
+        embeddings = []
+        for i in range(batch_size):
+            idx = inputs[i][0]
+            embeddings.append([idx] * 10)
+        
+        return torch.tensor(embeddings)
```

Comment:
I think we can use `Tensor.repeat` or something similar to make this nicer and more compact. Also, I would not make `forward` a static method.